docs/source/cpp/env_vars.rst

```diff
@@ -148,19 +152,29 @@ that changing their value later will have an effect.
    platforms currently do not implement any form of runtime dispatch.
 
    .. note::
-      In addition to runtime dispatch, the compile-time SIMD level can
-      be set using the ``ARROW_SIMD_LEVEL`` CMake configuration variable.
-      Unlike runtime dispatch, compile-time SIMD optimizations cannot be
-      changed at runtime (for example, if you compile Arrow C++ with AVX512
-      enabled, the resulting binary will only run on AVX512-enabled CPUs).
-      Setting ``ARROW_USER_SIMD_LEVEL=NONE`` prevents the execution of
-      explicit SIMD optimization code, but it does not rule out the execution
-      of compiler generated SIMD instructions.  E.g., on x86_64 platform,
-      Arrow is built with ``ARROW_SIMD_LEVEL=SSE4_2`` by default.  Compiler
-      may generate SSE4.2 instructions from any C/C++ source code.  On legacy
-      x86_64 platforms do not support SSE4.2, Arrow binary may fail with
-      SIGILL (Illegal Instruction).  User must rebuild Arrow and PyArrow from
-      scratch by setting cmake option ``ARROW_SIMD_LEVEL=NONE``.
+      In addition to runtime dispatch, Arrow's build system supports two
+      compile-time CMake variables for controlling SIMD: ``ARROW_SIMD_LEVEL``
+      and ``ARROW_RUNTIME_SIMD_LEVEL``.
+
+      The ``ARROW_SIMD_LEVEL`` variable sets the minimum supported SIMD level at
+      the compiler level. This works by passing the appropriate target
```

Comment:
This is a bit misleading IMHO. It sets which instructions the compiler is allowed to emit regardless of dynamic dispatch.

Reply:
Ah, I think I see now. I just took a closer look at how ARROW_SIMD_LEVEL gets handled and the options don't stack onto each other. Borrowing from your language, how's this?

```suggestion
      The ``ARROW_SIMD_LEVEL`` variable controls which SIMD instructions the compiler is allowed to emit. 
 This works by passing the appropriate target
```

Reply:
Hmm, this is still misleading in a different way. SIMD instructions are also emitted for runtime-selected optimizaitions, and they are not controlled by `ARROW_SIMD_LEVEL` :-)

Perhaps my suggestion was a dead end afterwards, and we shouldn't try to phrase this in terms of "emit[ting] SIMD instructions"?